pkg/recipes/configloader/secrets.go

```diff
@@ -55,33 +64,55 @@ func (e *secretsLoader) LoadSecrets(ctx context.Context, secretStoreIDResourceKe
 			return nil, err
 		}
 
-		// Populate the secretData map.
-		secretData, err = populateSecretData(secretStoreID, secretKeys, &secrets)
+		// Populate secretStoreData with secret type and map of secret keys and values
+		secretStoreData, err := populateSecretData(secretStoreID, secretKeysFilter, &secrets)
 		if err != nil {
 			return nil, err
 		}
+
+		secretData[secretStoreID] = secretStoreData
 	}
 
 	return secretData, nil
 }
 
 // populateSecretData is a helper function to populate secret data from a secret store.
-func populateSecretData(secretStoreID string, secretKeys []string, secrets *v20231001preview.SecretStoresClientListSecretsResponse) (map[string]map[string]string, error) {
-	secretData := make(map[string]map[string]string)
-
+// It takes a secret store ID, a filter for secret keys, and a response containing the secret data.
+// It returns SecretData{} populated with secret Type and a map with secret keys and their corresponding values.
+// ---
+// If the secret keys filter is nil or empty, it retrieves data for all keys in the secret store.
+// Eg: secretKeysFilter = nil or secretKeysFilter = [] will retrieve data for all secrets for secretStoreID.
+// ---
+// If the secret keys filter is populated, it retrieves data for the specified keys.
+// Eg: secretKeysFilter = [secretkey1] will retrieve data for only 'secretkey1' key.
+func populateSecretData(secretStoreID string, secretKeysFilter []string, secrets *v20231001preview.SecretStoresClientListSecretsResponse) (recipes.SecretData, error) {
 	if secrets == nil {
-		return nil, fmt.Errorf("secrets not found for secret store ID '%s'", secretStoreID)
+		return recipes.SecretData{}, fmt.Errorf("secrets not found for secret store ID '%s'", secretStoreID)
+	}
+
+	if secrets.Type == nil {
+		return recipes.SecretData{}, fmt.Errorf("secret store type is not set for secret store ID '%s'", secretStoreID)
+	}
+
+	secretData := recipes.SecretData{
+		Type: string(*secrets.Type),
+		Data: make(map[string]string),
+	}
+
+	// If secretKeysFilter is nil or empty, populate secretKeysFilter with all keys
+	if len(secretKeysFilter) == 0 {
+		secretKeysFilter = make([]string, 0, len(secrets.Data))
+		for secretKey := range secrets.Data {
+			secretKeysFilter = append(secretKeysFilter, secretKey)
+		}
 	}
```

Comment:
can we do it this way instead of creating a secretKeyFilter map ?
```suggestion
	if len(secretKeysFilter) == 0 {
	        secretData.Data = secrets.Data
	        return secretData,nil
	}
```

Reply:
secrets.Data is a map[string]*SecretValueProperties, different datatype from what is needed. keeping current code as discussed offline.